### PR TITLE
Bump up the thread priorities of the GPU, UI and IO threads on Android.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

**Comment by chinmaygarde** [comment]
Verified in systrace:
screen shot 2016-08-29 at 2 45 25 pm

**Comment by chinmaygarde** [comment]
cc @jason-simmons @abarth

File: sky/shell/platform/android/platform_view_android.cc [hunk 1]
```diff
@@ -401,6 +401,19 @@ void PlatformViewAndroid::SurfaceCreated(JNIEnv* env,
...
+
+void PlatformViewAndroid::UpdateThreadPriorities() {
+  blink::Threads::Gpu()->PostTask(
+      []() { ::setpriority(PRIO_PROCESS, gettid(), -3); });
+
+  blink::Threads::UI()->PostTask(
+      []() { ::setpriority(PRIO_PROCESS, gettid(), -2); });
+
+  blink::Threads::IO()->PostTask(
+      []() { ::setpriority(PRIO_PROCESS, gettid(), -1); });
```
Comment (abarth) [inline sky/shell/platform/android/platform_view_android.cc]: Do we need to set the priority of the IO thread?  It's supposed to be for bulk background work.

Comment (chinmaygarde) [inline sky/shell/platform/android/platform_view_android.cc]: In that case, I can leave the IO thread priority as is.

**Comment by abarth** [comment]
LGTM

Mucking with thread priorities is a tricky business.  It's worth a try though.

**Comment by chinmaygarde** [comment]
Yeah, @dragostis and I were tracing another perf issue and noticed that threads servicing logs, etc. were being scheduled when the GPU or UI thread were ready for work. This bumps us up in the queue. Its still quite conservative though.

**Comment by chinmaygarde** [comment]
Rebased.

**Pull request merged**